src/Type/Constant/ConstantArrayType.php

```diff
@@ -1012,6 +1006,120 @@ public function sliceArray(Type $offsetType, Type $lengthType, TrinaryLogic $pre
 		return $builder->getArray();
 	}
 
+	public function spliceArray(Type $offsetType, Type $lengthType, Type $replacementType): Type
+	{
+		$keyTypesCount = count($this->keyTypes);
+
+		$offset = $offsetType instanceof ConstantIntegerType ? $offsetType->getValue() : null;
+
+		if ($lengthType instanceof ConstantIntegerType) {
+			$length = $lengthType->getValue();
+		} elseif ($lengthType->isNull()->yes()) {
+			$length = $keyTypesCount;
+		} else {
+			$length = null;
+		}
+
+		if ($offset === null || $length === null) {
+			return $this->degradeToGeneralArray()
+				->spliceArray($offsetType, $lengthType, $replacementType);
+		}
+
+		if ($keyTypesCount + $offset <= 0) {
```

Comment:
I wonder whether thinks like this if-condition could be made more general with a `isSuperTypeOf` instead of a operation on `ConstantIntegerType` only

Reply:
I'd like to change this, but for the same reasons mentioned in https://github.com/phpstan/phpstan-src/pull/3952#discussion_r2052024142 I seem to can't be doing this right now.